src/openforms/forms/api/serializers/form_definition.py

```diff
@@ -131,6 +134,21 @@ def validate(self, attrs):
             validate_form_definition_is_reusable(
                 self.instance, new_value=attrs.get("is_reusable")
             )
+
+        # during import, process legacy format component translations
+        if self.context.get("is_import", False) and (
+            translations_store := attrs.get("component_translations")
+        ):
+            warnings.warn(
+                "Form-definition component translations are deprecated, the "
+                "compatibility layer wil be removed in Open Forms 3.0.",
+                DeprecationWarning,
+            )
```

Comment:
Maybe set a relevant `stacklevel` so that we can actually see where the serializer has been used (not sure if this is possible, this `validate` method is probably nested under a lot of calls)

Reply:
I don't think this has any added value, since the people importing legacy forms will not see this warning anyway. It's mostly a reminder for ourselves to remove this shim in Open Forms 3.0.